### PR TITLE
Feature cadastro dog

### PR DESCRIPTION
**Pull request by luizaboralli**

### CODE REVIEW

**Comment by ignaciojvig** [comment]
@luizaboralli, é necessário antes de tudo fazer um merge para trazer o conteudo da master para sua branch, pois existe conteúdo novo nela.

**Comment by ignaciojvig** [comment]
@luizaboralli também é necessario incluir seu endpoint no Swagger

File: adapters/api/routes/dog.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+const invalidRequestReply = (request, reply, errors) => reply.status(400).json({
+  method: request.method,
+  status: reply.statusCode,
+  error: errors,
+});
+
+module.exports = (app) => {
+  app.post('/dog', validators.registerValidator(), async (request, reply) => {
+    /*  #swagger.parameters['post dog object'] = {
```
Comment (ignaciojvig) [inline adapters/api/routes/dog.js:12]: Pode remover o comentário pois aqui ele pode atrapalhar a legibilidade do código

File: adapters/api/routes/dog.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    return reply.json(response);
+  });
+
+  app.get('/dog/:id', async (request, reply) => {
+    const response = await controller.getById(request.params.id, request, reply);
+    return reply.json(response);
+  });
+
+  app.put('/dog/:id', validators.updateValidator(), async (request, reply) => {
+    /*  #swagger.parameters['put dog object'] = {
```
Comment (ignaciojvig) [inline adapters/api/routes/dog.js:41]: Aqui também

**Comment by ignaciojvig** [comment]
@luizaboralli se puder, seria legal ter os models de dogs nos endpoints de POST e PUT do Swagger, pra conseguirmos testar eles com mais facilidade

**Comment by ignaciojvig** [comment]
@luizaboralli, encontrei um probleminha rolando no POST.
Conforme conversamos, aparenta ser devido a uma discrepancia de campos entre o model de Dogs e a Migration de Dogs
Log do POST via Swagger abaixo:
`
ValidationError [SequelizeValidationError]: notNull Violation: Dog.status cannot be null
    at InstanceValidator._validate (C:\Users\vntjoig\Projects\WomenCanCode\ong-manager-team1\node_modules\sequelize\lib\instance-validator.js:78:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async InstanceValidator._validateAndRunHooks (C:\Users\vntjoig\Projects\WomenCanCode\ong-manager-team1\node_modules\sequelize\lib\instance-validator.js:111:7)
    at async InstanceValidator.validate (C:\Users\vntjoig\Projects\WomenCanCode\ong-manager-team1\node_modules\sequelize\lib\instance-validator.js:93:12)
    at async Dog.save (C:\Users\vntjoig\Projects\WomenCanCode\ong-manager-team1\node_modules\sequelize\lib\model.js:3877:7)
    at async Function.create (C:\Users\vntjoig\Projects\WomenCanCode\ong-manager-team1\node_modules\sequelize\lib\model.js:2207:12)
    at async Object.exports.register (C:\Users\vntjoig\Projects\WomenCanCode\ong-manager-team1\application\services\dog.js:5:20)
    at async Object.exports.post (C:\Users\vntjoig\Projects\WomenCanCode\ong-manager-team1\adapters\controllers\dog.js:4:18)
    at async C:\Users\vntjoig\Projects\WomenCanCode\ong-manager-team1\adapters\api\routes\dog.js:17:22 {
  errors: [
    ValidationErrorItem {
      message: 'Dog.status cannot be null',
      type: 'notNull Violation',
      path: 'status',
      value: null,
      origin: 'CORE',
      instance: [Dog],
      validatorKey: 'is_null',
      validatorName: null,
      validatorArgs: []
    }
  ]
}
C:\Users\vntjoig\Projects\WomenCanCode\ong-manager-team1\application\services\dog.js:9
    const error = new Error('An error ocurred while creating a dog');
                  ^

Error: An error ocurred while creating a dog
    at Object.exports.register (C:\Users\vntjoig\Projects\WomenCanCode\ong-manager-team1\application\services\dog.js:9:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Object.exports.post (C:\Users\vntjoig\Projects\WomenCanCode\ong-manager-team1\adapters\controllers\dog.js:4:18)
    at async C:\Users\vntjoig\Projects\WomenCanCode\ong-manager-team1\adapters\api\routes\dog.js:17:22 {
  statusCode: 500
}
`

File: adapters/api/routes/dog.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+const invalidRequestReply = (request, reply, errors) => reply.status(400).json({
+  method: request.method,
+  status: reply.statusCode,
+  error: errors,
+});
+
+module.exports = (app) => {
+  app.post('/dog', validators.registerValidator(), async (request, reply) => {
+
```
Comment (ego-giova) [inline adapters/api/routes/dog.js:12]: Adicionar os comentários do Swagger igual fizemos para o Vet;

File: adapters/api/routes/dog.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    return reply.json(response);
+  });
+
+  app.get('/dog/:id', async (request, reply) => {
+    const response = await controller.getById(request.params.id, request, reply);
+    return reply.json(response);
+  });
+
+  app.put('/dog/:id', validators.updateValidator(), async (request, reply) => {
+
```
Comment (ego-giova) [inline adapters/api/routes/dog.js:32]: Adicionar os comentários do Swagger igual fizemos para o Vet;

**Review by ego-giova — CHANGES_REQUESTED** [review]

**Review by ego-giova — APPROVED** [review]

**Pull request merged**